### ISSUE TITLE
Waypoints Crash

### ISSUE THREAD

**Issue by Kolobok125040** [SkyblockerMod/Skyblocker#677]
### Minecraft Version

1.20.4

### Skyblocker Version

skyblocker-1.20.2+1.20.4.jar

### Reproduction Steps

I do:
1. `/skyblocker waypoints ordered add eggs 494 106 556`
2. `/skyblocker waypoints ordered add eggs 510 92 470`
3. `/skyblocker waypoints ordered removeAt eggs 1`
4. `/skyblocker waypoints ordered removeAt eggs 0`

Than game crashed

### Crash Report file and latest.log

```shell
---- Minecraft Crash Report ----
// This doesn't make any sense!

Time: 2024-04-29 14:33:58
Description: Unexpected error

java.lang.ArithmeticException: / by zero
	at de.hysky.skyblocker.skyblock.waypoint.OrderedWaypoints.render(OrderedWaypoints.java:238)
	at net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents.lambda$static$22(WorldRenderEvents.java:222)
	at net.minecraft.class_761.handler$bee000$fabric-rendering-v1$beforeClouds(class_761.java:5798)
	at net.minecraft.class_761.method_22710(class_761.java:1283)
	at net.minecraft.class_757.method_3188(class_757.java:1098)
	at net.minecraft.class_757.method_3192(class_757.java:866)
	at net.minecraft.class_310.method_1523(class_310.java:1327)
	at net.minecraft.class_310.method_1514(class_310.java:888)
	at net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.20.4
	Minecraft Version ID: 1.20.4
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.7, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 1231026176 bytes (1173 MiB) / 2952790016 bytes (2816 MiB) up to 4294967296 bytes (4096 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: 11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz
	Identifier: Intel64 Family 6 Model 141 Stepping 1
	Microarchitecture: unknown
	Frequency (GHz): 2.69
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: Intel(R) UHD Graphics
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MB): 128.00
	Graphics card #0 deviceId: 0x9a68
	Graphics card #0 versionInfo: DriverVersion=31.0.101.5382
	Graphics card #1 name: NVIDIA GeForce RTX 3050 Laptop GPU
	Graphics card #1 vendor: NVIDIA (0x10de)
	Graphics card #1 VRAM (MB): 4095.00
	Graphics card #1 deviceId: 0x25a2
	Graphics card #1 versionInfo: DriverVersion=31.0.15.5222
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 3.20
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 8192.00
	Memory slot #1 clockSpeed (GHz): 3.20
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 27349.67
	Virtual memory used (MB): 16887.65
	Swap memory total (MB): 11264.00
	Swap memory used (MB): 166.81
	JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms512m -Xmx4096m
	Fabric Mods: 
		aaron-mod: Aaron's Mod 1.2.2
			me_nullicorn_nedit: Nedit 2.2.0
		architectury: Architectury 11.1.17
		chatpatches: Chat Patches 204.6.4
		citresewn: CIT Resewn 1.1.4+1.20.4
			citresewn-defaults: CIT Resewn: Defaults 1.1.4+1.20.4
		cloth-config: Cloth Config v13 13.0.121
			cloth-basic-math: cloth-basic-math 0.6.1
		continuity: Continuity 3.0.0-beta.4+1.20.2
		cubes-without-borders: Cubes Without Borders 1.1.0+1.20
		dulkirmod-fabric: DulkirMod-Fabric 1.0.3
			jarviscci: Jarvis Common Config Index 1.1.1
			meteordevelopment_orbit: orbit 0.2.3
		dynamic_fps: Dynamic FPS 3.4.3
		edgelesschatscreen: Edgeless Chat Screen 1.1.2
		elementa: Elementa 629
		entityculling: EntityCulling 1.6.4
		essential: Essential 1.3.1.3+g88238d7752
		essential-container: essential-container 1.0.0
			essential-loader: essential-loader 1.2.1
		fabric-api: Fabric API 0.96.11+1.20.4
			fabric-api-base: Fabric API Base 0.4.36+78d798af4f
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.49+82b1bb3e4f
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.16+78d798af4f
			fabric-block-api-v1: Fabric Block API (v1) 1.0.16+3e2216cb4f
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.4+78d798af4f
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.46+78d798af4f
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.7+78d798af4f
			fabric-command-api-v1: Fabric Command API (v1) 1.2.41+f71b366f4f
			fabric-command-api-v2: Fabric Command API (v2) 2.2.20+78d798af4f
			fabric-commands-v0: Fabric Commands (v0) 0.2.58+df3654b34f
			fabric-containers-v0: Fabric Containers (v0) 0.1.86+df3654b34f
			fabric-content-registries-v0: Fabric Content Registries (v0) 5.0.15+78d798af4f
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.10+78d798af4f
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.23+78d798af4f
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.1.4+b90db5744f
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 13.2.4+5c0133444f
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.61+78d798af4f
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.6.1+09fc25014f
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.7.1+389931eb4f
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.74+df3654b34f
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.46+78d798af4f
			fabric-item-api-v1: Fabric Item API (v1) 2.2.0+d6f2b0844f
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.26+58f8c0124f
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.41+78d798af4f
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.39+df3654b34f
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.3.0+a67ffb5d4f
			fabric-loot-api-v2: Fabric Loot API (v2) 2.1.9+78d798af4f
			fabric-message-api-v1: Fabric Message API (v1) 6.0.5+78d798af4f
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.65+78d798af4f
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 1.0.8+78d798af4f
			fabric-models-v0: Fabric Models (v0) 0.4.7+9386d8a74f
			fabric-networking-api-v1: Fabric Networking API (v1) 3.1.7+2e5ac5484f
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 13.0.14+080016e44f
			fabric-particles-v1: Fabric Particles (v1) 1.1.7+78d798af4f
			fabric-recipe-api-v1: Fabric Recipe API (v1) 2.0.20+78d798af4f
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 4.0.19+58f8c0124f
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.2.4+78d798af4f
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.5.4+78d798af4f
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.53+df3654b34f
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.42+73761d2e4f
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.1.1+e761c6694f
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.56+df3654b34f
			fabric-rendering-v1: Fabric Rendering (v1) 3.2.0+6fd945a04f
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.14+78d798af4f
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.20+df798a894f
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.17+78d798af4f
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.55+78d798af4f
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.17+78d798af4f
			fabric-transfer-api-v1: Fabric Transfer API (v1) 4.0.11+eb30349a4f
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 5.0.14+78d798af4f
		fabric-language-kotlin: Fabric Language Kotlin 1.10.19+kotlin.1.9.23
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.9.23
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.9.23
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.9.23
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.9.23
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.23.2
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.8.0
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.8.0
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.5.0
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.6.3
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.6.3
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.6.3
		fabricloader: Fabric Loader 0.15.10
			mixinextras: MixinExtras 0.3.5
		ferritecore: FerriteCore 6.0.3
		firmament: Firmament 1.4.0
			io_ktor_ktor-client-content-negotiation-jvm: ktor-client-content-negotiation-jvm 2.3.0
			io_ktor_ktor-client-core-jvm: ktor-client-core-jvm 2.3.0
			io_ktor_ktor-client-encoding-jvm: ktor-client-encoding-jvm 2.3.0
			io_ktor_ktor-client-java-jvm: ktor-client-java-jvm 2.3.0
			io_ktor_ktor-client-logging-jvm: ktor-client-logging-jvm 2.3.0
			io_ktor_ktor-events-jvm: ktor-events-jvm 2.3.0
			io_ktor_ktor-http-jvm: ktor-http-jvm 2.3.0
			io_ktor_ktor-io-jvm: ktor-io-jvm 2.3.0
			io_ktor_ktor-serialization-jvm: ktor-serialization-jvm 2.3.0
			io_ktor_ktor-serialization-kotlinx-json-jvm: ktor-serialization-kotlinx-json-jvm 2.3.0
			io_ktor_ktor-serialization-kotlinx-jvm: ktor-serialization-kotlinx-jvm 2.3.0
			io_ktor_ktor-utils-jvm: ktor-utils-jvm 2.3.0
			io_ktor_ktor-websocket-serialization-jvm: ktor-websocket-serialization-jvm 2.3.0
			io_ktor_ktor-websockets-jvm: ktor-websockets-jvm 2.3.0
			libgui: LibGui 9.2.2+1.20.2
				jankson: Jankson 6.0.0+j1.2.3
					blue_endless_jankson: jankson 1.2.3
				libninepatch: LibNinePatch 1.2.0
			mm_shedaniel: Manningham Mills (shedaniel's fork) 2.4.1
			moulconfig: moulconfig 1.0.0
			org_slf4j_slf4j-api: slf4j-api 1.7.36
		forcecloseworldloadingscreen: kennytv's epic force close loading screen mod for Fabric 2.2.1
		gamma_shifter: Gamma Shifter 1.0.4+mc1.20.4
		immediatelyfast: ImmediatelyFast 1.2.12+1.20.4
			net_lenni0451_reflect: Reflect 1.3.2
		indium: Indium 1.0.30+mc1.20.4
		inventoryhud: Inventory HUD + 3.4.19
		iris: Iris 1.7.0+mc1.20.4
			io_github_douira_glsl-transformer: glsl-transformer 2.0.0-pre13
			org_anarres_jcpp: jcpp 1.4.14
			org_antlr_antlr4-runtime: antlr4-runtime 4.11.1
		java: Java HotSpot(TM) 64-Bit Server VM 17
		krypton: Krypton 0.2.6
			com_velocitypowered_velocity-native: velocity-native 3.2.0-SNAPSHOT
		libjf: LibJF 3.14.3
			libjf-base: LibJF Base 3.14.3
			libjf-config-commands: LibJF Config Commands 3.14.3
			libjf-config-core-v2: LibJF Config 3.14.3
			libjf-config-network-v0: LibJF Config: Network 3.14.3
			libjf-config-ui-tiny: LibJF Config UI: Tiny 3.14.3
			libjf-data-manipulation-v0: LibJF Data Manipulation 3.14.3
			libjf-data-v0: LibJF Data 3.14.3
			libjf-mainhttp-v0: LibJF MainHTTP 3.14.3
			libjf-translate-v1: LibJF Translate 3.14.3
			libjf-unsafe-v0: LibJF Unsafe 3.14.3
			libjf-web-v1: LibJF Web 3.14.3
		lithium: Lithium 0.12.1
		malilib: MaLiLib 0.18.1
		minecraft: Minecraft 1.20.4
		modmenu: Mod Menu 9.2.0-beta.2
		nbttooltip: NBT Tooltip 1.8.1
		notenoughcrashes: Not Enough Crashes 4.4.7+1.20.4
		optigui: OptiGUI 2.3.0-beta.2
			org_apache_commons_commons-text: commons-text 1.11.0
			org_ini4j_ini4j: ini4j 0.5.4
		org_jetbrains_annotations: annotations 13.0
		reeses-sodium-options: Reese's Sodium Options 1.7.2+mc1.20.4-build.102
		roughlyenoughitems: Roughly Enough Items 14.1.720
			error_notifier: Error Notifier 1.0.9
		skinlayers3d: 3d-Skin-Layers 1.6.3
		skyblocker: Skyblocker 1.20.2+1.20.4
			com_logisticscraft_occlusionculling: occlusionculling 0.0.8-SNAPSHOT
			meteordevelopment_discord-ipc: discord-ipc 1.1
			moe_nea_neurepoparser: neurepoparser 1.4.0
		skyblockmod: Skyblock Mod 1.10.0
			org_apache_commons_commons-math3: commons-math3 3.6.1
			org_eclipse_jgit_org_eclipse_jgit: org.eclipse.jgit 6.8.0.202311291450-r
			renderer: Renderer 1.1
		slyde: Slyde 1.7.3
		sodium: Sodium 0.5.8+mc1.20.4
		sodium-extra: Sodium Extra 0.5.4+mc1.20.4-build.116
			caffeineconfig: CaffeineConfig 1.3.0+1.17
			crowdin-translate: CrowdinTranslate 1.4+1.19.3
		tooltipscroll: Tooltip Scroll 1.3.0
		tweakeroo: Tweakeroo 0.19.2
		universalcraft: UniversalCraft 323
		vigilance: Vigilance 295
		yet_another_config_lib_v3: YetAnotherConfigLib 3.3.2+1.20.4
			com_twelvemonkeys_common_common-image: common-image 3.10.0
			com_twelvemonkeys_common_common-io: common-io 3.10.0
			com_twelvemonkeys_common_common-lang: common-lang 3.10.0
			com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0
			com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0
			com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0
			org_quiltmc_parsers_gson: gson 0.2.1
			org_quiltmc_parsers_json: json 0.2.1
		zoomify: Zoomify 2.13.2
			com_akuleshov7_ktoml-core-jvm: ktoml-core-jvm 0.5.1
			dev_isxander_settxi_settxi-core: settxi-core 2.10.6
			dev_isxander_settxi_settxi-kotlinx-serialization: settxi-kotlinx-serialization 2.10.6
	Loaded Shaderpack: (off)
	Client Crashes Since Restart: 1
	Integrated Server Crashes Since Restart: 0
	Suspected Mods: Fabric Rendering (v1) (fabric-rendering-v1), Skyblocker (skyblocker)
```

### Additional context

![изображение](https://github.com/SkyblockerMod/Skyblocker/assets/60614259/0a5b0b43-365c-424f-87ed-4988e1b7ee9a)

**Comment by Kolobok125040** [comment]
And after that all particles in game was black squares

**Comment by viciscat** [comment]
I know why it crashes. Do these other issues happen after it has "crashed" (I see that you have Not enough crashes)? Cuz I can't reproduce em

**Comment by viciscat** [comment]
try this fix https://nightly.link/SkyblockerMod/Skyblocker/actions/artifacts/1458127836.zip
don't forget to unzip!

**Comment by Kolobok125040** [comment]
It don't crash now, but i tested it with out other mods because I don't want update them to 1.20.5

**Comment by AzureAaron** [comment]
Fixed by #669
